cli/src/package.rs

```diff
@@ -1,36 +1,135 @@
-use std::{env::current_dir, fs, path::PathBuf};
-
+use crate::error::CliError;
 use directories::ProjectDirs;
+use futures::future::join_all;
+use modmark_core::Resolve;
+use serde_json;
+use std::{
+    env::current_dir,
+    fs::{self, create_dir_all, File},
+    io::copy,
+    path::PathBuf,
+};
 
-pub trait Resolve {
-    fn resolve(&self, path: &str) -> Option<Vec<u8>>;
-}
+static RUNTIME: once_cell::sync::Lazy<tokio::runtime::Runtime> = once_cell::sync::Lazy::new(|| {
+    tokio::runtime::Builder::new_multi_thread()
+        .enable_all()
+        .build()
+        .unwrap()
+});
+
+pub struct PackageManager {}
 
-pub struct PackageManager;
 impl Resolve for PackageManager {
-    fn resolve(&self, path: &str) -> Option<Vec<u8>> {
-        let mut local_path = match current_dir() {
-            Ok(p) => p,
-            Err(_) => return None,
-        };
+    type Error = CliError;
+    fn resolve(&self, path: &str) -> Result<Vec<u8>, Self::Error> {
+        let out = RUNTIME.block_on(resolve_package(path))?;
```

Comment:
We don't need to bind this to a variable, we can return it directly
```suggestion
        RUNTIME.block_on(resolve_package(path))
```